openml/setups/functions.py

```diff
@@ -74,6 +74,76 @@ def get_setup(setup_id):
     return _create_setup_from_xml(result_dict)
 
 
+def setup_list(flow=None, tag=None, offset=None, size=None):
+    """List all setups matching all of the given filters.
+
+        Perform API call `/setup/list/{filters}
+
+        Parameters
+        ----------
+        flow : int, optional
+
+        tag : str, optional
+
+        offset : int, optional
+
+        size : int, optional
+
+        Returns
+        -------
+        list
+            List of found setups.
+        """
+
+    api_call = "setup/list"
+    if offset is not None:
+        api_call += "/offset/%d" % int(offset)
+    if size is not None:
+        api_call += "/limit/%d" % int(size)
+    if flow is not None:
+        api_call += "/flow/%s" % flow
+    if tag is not None:
+        api_call += "/tag/%s" % tag
+
+    return _list_setups(api_call)
+
+
+def _list_setups(api_call):
+    """Helper function to parse API calls which are lists of setups"""
+
+    xml_string = openml._api_calls._perform_api_call(api_call)
+
+    setups_dict = xmltodict.parse(xml_string)
+    # Minimalistic check if the XML is useful
+    if 'oml:setups' not in setups_dict:
+        raise ValueError('Error in return XML, does not contain "oml:setups": %s'
+                         % str(setups_dict))
+    elif '@xmlns:oml' not in setups_dict['oml:setups']:
+        raise ValueError('Error in return XML, does not contain '
+                         '"oml:runs"/@xmlns:oml: %s'
+                         % str(setups_dict))
+    elif setups_dict['oml:setups']['@xmlns:oml'] != 'http://openml.org/openml':
+        raise ValueError('Error in return XML, value of  '
+                         '"oml:runs"/@xmlns:oml is not '
+                         '"http://openml.org/openml": %s'
+                         % str(setups_dict))
+
+    if isinstance(setups_dict['oml:setups']['oml:setup'], list):
+        setups_list = setups_dict['oml:setups']['oml:setup']
+    elif isinstance(setups_dict['oml:setups']['oml:setup'], dict):
+        setups_list = [setups_dict['oml:setups']['oml:setup']]
+    else:
+        raise TypeError()
+
+    setups = dict()
+    for setup_ in setups_list:
+        # making it a dict to give it the right format
+        current = _create_setup_from_xml({'oml:setup_parameters': setup_})
```

Comment:
I see your point, but in this sense this is more rich, right? 

This listing function returns all information available, whereas other functions show just a summary of the function. There is no point in throwing this away